src/secret_service_with_keyutils.rs

```diff
@@ -0,0 +1,169 @@
+/*!
+
+# secret-service-with-keyutils credential store
+
+TODO
+
+ */
+use std::collections::HashMap;
+
+#[cfg(feature = "sync-secret-service")]
+use dbus_secret_service::{Error, Item};
+#[cfg(feature = "async-secret-service")]
+use secret_service::{blocking::Item, Error};
+
+#[cfg(all(target_os = "linux", feature = "linux-native"))]
+use crate::keyutils::KeyutilsCredential;
+use crate::secret_service::{decode_error, SsCredential};
+
+use super::credential::{Credential, CredentialApi, CredentialBuilder, CredentialBuilderApi};
+use super::error::Result;
+
+#[derive(Debug, Clone)]
+pub struct SsKeyutilsCredential {
+    keyutils: KeyutilsCredential,
+    ss: SsCredential,
+}
+
+impl CredentialApi for SsKeyutilsCredential {
+    fn set_password(&self, password: &str) -> Result<()> {
+        self.set_secret(password.as_bytes())
+    }
+
+    fn set_secret(&self, secret: &[u8]) -> Result<()> {
+        self.ss.set_secret(secret)?;
+        let _ = self.keyutils.set_secret(secret);
```

Comment:
You cannot ignore this error, because there are secrets that are acceptable to secret service (such as an empty secret) that are not acceptable to keyutils. If your service is going to cache in keyutils, it should only accept secrets that can be cached in keyutils. I would recommend that you invert the logic here: first store in keyutils, and then if you don't get an error store in secret service.